dht_test.go

```diff
@@ -1819,110 +1819,6 @@ func TestDynamicModeSwitching(t *testing.T) {
 	assertDHTClient()
 }
 
-func TestProtocolUpgrade(t *testing.T) {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
```

Comment:
This test no longer seems necessary. Although we may want to resurrect something like it in the future